Add 'update' parameter in junos_config module

ganeshrn opened the pull request:
##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Feature Pull Request

##### COMPONENT NAME
junos_config

##### ANSIBLE VERSION
<!---
Paste verbatim output from “ansible --version” between quotes below,
this is to help the Ansible team determine if this is a version specific
issue which is being fixed.
-->
```
ansible --version
ansible 2.3.0 
```

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

<!---
If you are fixing an existing issue, please include "Fixes #nnnn" in your commit
message and your description; but you should still explain what the change does.
-->

<!---
Please paste verbatim command output below, e.g. before and after your change.
Even in the event of a Docs Pull Request, this allows the Ansible team to quickly
verify that there were no accidental syntax mistakes.
-->
```
Add update parameter in junos_config module which supports
configuration action like merge, replace and overwrite.
```

Related PR https://github.com/ansible/ansible/pull/19046
Fixes issue: https://github.com/ansible/ansible-modules-core/issues/5839

ganeshrn committed:
Fix # #5839 Add 'update' parameter in junos_config module

Add update parameter in junos_config module which supports
configuration action like merge, replace and overwrite.

ganeshrn committed:
Fix documentation issue

ganeshnalawade commented:
Migrated from https://github.com/ansible/ansible-modules-core/pull/5855

gundalow reviewed (changes requested):
This removes function that existed in the 2.2 release, so it will need to be discussed more, and possibly an `alias:` option added. Please add to the Networking Group Agenda and we can discuss there.

ganeshrn committed:
Fix review comment to add replace argument

Make replace and update argument mutually
exclusive, to support replace for backward
compatibility.

ganeshnalawade commented:
@privateip @gundalow @Qalthos As per discussion in Networking group meeting (16th Dec 2106) I have added support for **replace** argument as well for backward compatibility.  **replace** and the new **update** argument is made mutually exclusive.  This PR and https://github.com/ansible/ansible/pull/19046 ensures existing functionality of **replace** works as it is and also adds support for new **update** argument.
